source/lex.h

```diff
@@ -333,6 +333,15 @@ auto labelized_position(token const* t)
     return ret;
 }
 
+auto unnamed_type_param_name(int ordinal, token const* t)
+    -> std::string
+{
+    return "UnnamedTypeParam"
+            + std::to_string(ordinal)
+            + "_"
+            + labelized_position(t);
```

Comment:
> Because template parameter names must be unique,

That's what ordinal numbering addresses for all parameters.

We only use labelized positions inside function bodies for things that don't already have a unique ordinal, like local unnnamed "auto" variables (and named `break`/`continue`, which IIRC was just in case there were two loops in the same function in non-nested `{` `}` that used the same loop name, to make sure we can generate unique jump target labels). But we could generate unique ordinals for those inside each function too, instead of using file/line positions... and maybe I should switch to that, because it would be more diff-stable as the file is changed.

Reply:
> But we could generate unique ordinals inside each function for those too, instead of using line/col positions... and maybe I should switch to that, because it would be more diff-stable as the file is changed.

Done: 7637c5001d8c7faff62549a5efe9afe1f9074ad9

Reply:
Ah, now I see the problem... template parameter names are not fully localized, and when there are nested template lists they don't share the same ordinal and need to be distinguished.

Thanks! It was still good to do the other cleanup to not rely on line/col names though.